roles/keystone/tasks/openidc.yml

```diff
@@ -26,3 +26,45 @@
 - name: enable apache mod auth_openidc
   apache2_module: name=auth_openidc
   notify: reload apache
+
+- name: create metadata folder for multiple oidc configuration files
+  file:
+    path: "/etc/apache2/openidc/metadata/"
+    state: directory
+    owner: www-data
+    group: root
+    mode: 0775
+    recurse: yes
+
+- name: create .client file for the IdPs
+  template:
+    src: etc/apache2/openidc/metadata/template.client
+    dest: /etc/apache2/openidc/metadata/{{ item.issuer | regex_replace('https://','') }}.client
+    owner: www-data
+    group: root
+    mode: 0640
+  with_items: "{{ keystone.federation.sp.oidc.providers_info }}"
+
+- name: create .conf file for the IdPs
+  template:
+    src: etc/apache2/openidc/metadata/template.conf
+    dest: /etc/apache2/openidc/metadata/{{ item.issuer | regex_replace('https://','') }}.conf
+    owner: www-data
+    group: root
+    mode: 0640
+  with_items: "{{ keystone.federation.sp.oidc.providers_info }}"
+
+- name: create .provider file for the IdPs
+  template:
+    src: etc/apache2/openidc/metadata/template.provider
+    dest: /etc/apache2/openidc/metadata/{{ item.issuer | regex_replace('https://','') }}.provider
+    owner: www-data
+    group: root
+    mode: 0640
+  when: item.provider_metadata_url is undefined
+  with_items: "{{ keystone.federation.sp.oidc.providers_info }}"
+
+- name: create .provider file for the IdPs
+  shell: "curl {{ item.provider_metadata_url }}  > /etc/apache2/openidc/metadata/{{ item.issuer | regex_replace('https://','') }}.provider"
```

Comment:
Mostly all IDP have an metadata_url. Since Bluebox Panel doesnt have one I check if the metadata variable is set and there are two flows from there